src/goto-instrument/contracts/contracts.cpp

```diff
@@ -396,7 +396,7 @@ void code_contractst::replace_old_parameter(
 
     if(
       parameter.id() == ID_dereference || parameter.id() == ID_member ||
-      parameter.id() == ID_symbol)
+      parameter.id() == ID_symbol || parameter.id() == ID_ptrmember)
```

Comment:
@SaswatPadhi to address your comment from https://github.com/diffblue/cbmc/pull/6277#discussion_r684679524

Reply:
Thanks! As you mentioned on the other PR, everything still works even with just `ID_member`, but I'm still not sure how. Let's leave `ID_ptrmember` in here as well. Do we allow these 4 cases in assigns clause as well?

Reply:
> Do we allow these 4 cases in assigns clause as well?
Yes, we do.

https://github.com/diffblue/cbmc/blob/a7bfc101b9ad1b95c755a368d6eda84317e565f5/src/ansi-c/ansi_c_convert_type.cpp#L251-L266